types/src/transaction/webauthn.rs

```diff
@@ -0,0 +1,693 @@
+        match (&public_key, &self.signature) {
+            (
+                AnyPublicKey::P256Ecdsa { public_key },
+                WebAuthnSignature::P256Ecdsa { signature },
+            ) => signature.verify_arbitrary_msg(&verification_data, public_key),
+            _ => Err(anyhow!(
+                "WebAuthn verification failure, invalid key, signature pairing"
+            )),
+        }
+    }
+}
+
+impl TryFrom<&[u8]> for PartialAuthenticatorAssertionResponse {
+    type Error = CryptoMaterialError;
+
+    /// Uses BCS serialization
+    fn try_from(
+        bytes: &[u8],
+    ) -> core::result::Result<PartialAuthenticatorAssertionResponse, CryptoMaterialError> {
+        bcs::from_bytes::<PartialAuthenticatorAssertionResponse>(bytes)
+            .map_err(|_e| CryptoMaterialError::DeserializationError)
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use crate::{
+        test_helpers::transaction_test_helpers::get_test_raw_transaction,
+        transaction::{
+            authenticator::{AnyPublicKey, AuthenticationKey},
+            webauthn::{PartialAuthenticatorAssertionResponse, WebAuthnSignature},
+            RawTransaction,
+        },
+    };
+    use anyhow::anyhow;
+    use aptos_crypto::{
+        p256_ecdsa,
+        p256_ecdsa::{P256PrivateKey, P256PublicKey, P256Signature},
+        signing_message, HashValue, PrivateKey, Uniform, ValidCryptoMaterialStringExt,
+    };
+    use coset::CoseKey;
+    use move_core_types::account_address::AccountAddress;
+    use p256::pkcs8::DecodePublicKey;
+    use passkey_authenticator::{
+        public_key_der_from_cose_key, user_validation::UserValidationMethod, Authenticator,
+    };
+    use passkey_client::Client;
+    use passkey_types::{
+        ctap2,
+        ctap2::Aaguid,
+        rand::random_vec,
+        webauthn::{
+            AttestationConveyancePreference, AuthenticatedPublicKeyCredential, CollectedClientData,
+            CreatedPublicKeyCredential, CredentialCreationOptions, CredentialRequestOptions,
+            PublicKeyCredentialCreationOptions, PublicKeyCredentialParameters,
+            PublicKeyCredentialRequestOptions, PublicKeyCredentialRpEntity,
+            PublicKeyCredentialType, PublicKeyCredentialUserEntity, UserVerificationRequirement,
+        },
+        Bytes, Passkey,
+    };
+    use rand::{rngs::StdRng, SeedableRng};
+    use url::Url;
+
+    type Challenge = Vec<u8>;
+    type RawTxnSigningMessage = Vec<u8>;
+
+    struct MyUserValidationMethod {}
+    #[async_trait::async_trait]
+    impl UserValidationMethod for MyUserValidationMethod {
+        async fn check_user_verification(&self) -> bool {
+            true
+        }
+
+        async fn check_user_presence(&self) -> bool {
+            true
+        }
+
+        fn is_presence_enabled(&self) -> bool {
+            true
+        }
+
+        fn is_verification_enabled(&self) -> Option<bool> {
+            Some(true)
+        }
+    }
+
```

Comment:
Explaining how these test cases can be reproduced would be helpful for future us.